core/src/main/java/hudson/slaves/NodeProvisioner.java

```diff
@@ -817,4 +828,77 @@ private static float getFloatSystemProperty(String propName, float defaultValue)
             }
         return defaultValue;
     }
+
+    private static void fireOnFailure(final NodeProvisioner.PlannedNode plannedNode, final Throwable cause) {
+        for (CloudProvisioningListener cl : CloudProvisioningListener.all()) {
+            try {
+                cl.onFailure(plannedNode, cause);
+            } catch (Throwable e) {
```

Comment:
`} catch (Error e) { throw E; `

Otherwise the LOGGER.log statement could re-trigger an OOM error)